internal/io/io.go

```diff
@@ -131,3 +131,29 @@ func ParseMediaType(content []byte) (string, error) {
 	}
 	return manifest.MediaType, nil
 }
+
+// ReadLine reads a line from the reader with trailing \r dropped.
+func ReadLine(reader io.Reader) ([]byte, error) {
+	line := make([]byte, 0)
+	drop := 0
+	for b := [1]byte{}; ; {
+		_, err := reader.Read(b[:])
+		if err != nil {
+			if err == io.EOF {
+				drop = 0
+				break
+			}
+			return nil, err
+		}
+		s := string(b[:])
+		if s == "\r" {
+			drop = 1
+		} else if s == "\n" {
+			break
+		} else {
+			drop = 0
+		}
+		line = append(line, b[0])
+	}
+	return line[:len(line)-drop], nil
```

Comment:
The implementation can be simplified as follows.
```suggestion
	var line []byte
	var buffer [1]byte
	for {
		n, err := reader.Read(b[:])
		if err != nil {
			if err == io.EOF {
				return line, nil
			}
			return nil, err
		}
		if n == 0 {
			continue
		}
		switch c := buffer[0]; c {
		case '\r', '\n':
			return line, nil
		default:
			line = append(line, c)
		}
	}
```

Reply:
1) according to https://pkg.go.dev/os#File.Read, if we are not doing zero-byte read, the returned read length `n` can only be 0 when EOF, why bother checking?
2) There is an issue in this implementation: ReadLine(strings.NewReader(\`abcde\r\n\`)) will leave unread `\n` in stdin.

Changed without including the above two

Reply:
Discussed offline, will change according to the below UX:
1. string ends CRLF: read until LF, remove CR and return left string
2. string ends with CR and EOF: remove CR and return left string
3. string contains CR before the last character: return string directly